### PR TITLE
qemu_guest_agent: Get hostname after change it in guest

### PR DESCRIPTION
**Pull request by 6-dehan**
Add steps to setting new hostname and check it in guest.

ID: 1853556
Signed-off-by: Dehan Meng \<demeng@redhat.com\>

### CODE REVIEW

**Comment by xiagao** [comment]
Pls add description ,bug id and signoff of this patch

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 1]
```diff
@@ -333,6 +333,7 @@
             no RHEL.6, RHEL.7
             gagent_check_type = os_basic_info
             cmd_get_host_name = hostname
+            cmd_set_host_name = "hostnamectl set-hostname; cat /etc/hostname"
```
Comment (xiagao) [inline qemu/tests/cfg/qemu_guest_agent.cfg:336]: what name do you change?
hostnamectl set-hostname xxx

Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg]: sorry,left it ,adding.

**Comment by xiagao** [comment]
Could you add test results?

**Comment by 6-dehan** [comment]
> Could you add test results?

you mean the test log that got a successful result?to where?

**Comment by 6-dehan** [comment]
Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_basic_info: PASS (110.88 s)

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -2963,7 +2963,7 @@ def gagent_check_os_basic_info(self, test, params, env):
         6) Check every user's domain(windows only)
         7) Get the earlier loggin time for the same user.
         8) Check the login time for every user.
-
+        9) Check host name after setting new host name
```
Comment (xiagao) [inline qemu/tests/qemu_guest_agent.py:2966]: could you add the step 9 to 'after' step 1

**Comment by 6-dehan** [comment]
okay，sure

**Comment by xiagao** [comment]
why there is four commits?
Do you use "git commit --amend"  when you modify your patch?

**Comment by 6-dehan** [comment]
@xiagao

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 3]
```diff
@@ -333,6 +333,7 @@
             no RHEL.6, RHEL.7
             gagent_check_type = os_basic_info
             cmd_get_host_name = hostname
+            cmd_set_host_name = "hostnamectl set-hostname newhostname; cat /etc/hostname"
```
Comment (xiagao) [inline qemu/tests/cfg/qemu_guest_agent.cfg:336]: if it's better to use replace ';' with '&&'
cmd_set_host_name = "hostnamectl set-hostname newhostname && cat /etc/hostname"

**Review by xiagao — APPROVED** [review]
LGTM
ack.

**Comment by 6-dehan** [comment]
@vivianQizhu Could you please have a check? thanks a lot.

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 4]
```diff
@@ -333,6 +333,7 @@
             no RHEL.6, RHEL.7
             gagent_check_type = os_basic_info
             cmd_get_host_name = hostname
+            cmd_set_host_name = hostnamectl set-hostname newhostname && cat /etc/hostname
```
Comment (vivianQizhu) [inline qemu/tests/cfg/qemu_guest_agent.cfg:336]: Is this works on Windows guest?

Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg:336]: thanks for reviewing.
this patch was fixed linux bug and we covered it into polarion ,but there are no steps about windows_guest situation,so I didn't consider windows guest cmd here.

Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg:336]: I‘ll add 'if' condition to filter linux os.what do you think?

Comment (vivianQizhu) [reply qemu/tests/cfg/qemu_guest_agent.cfg:336]: @6-dehan Yes please, otherwise it might fail on Windows guest.

Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg:336]: @vivianQizhu 
If condition has been updated,check please.

**Review by vivianQizhu — APPROVED** [review]

**Comment by vivianQizhu** [comment]
Thanks @6-dehan @xiagao

**Comment by 6-dehan** [comment]
> Thanks @6-dehan @xiagao

thank you for your reviewing and valuable suggestions.@xiagao,@vivianQizhu

**Pull request merged**